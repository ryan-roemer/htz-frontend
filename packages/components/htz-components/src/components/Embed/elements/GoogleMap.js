/* *************************************************************** *
 * This element accepts these inputTemplates:
  [
    com.polobase.GoogleMapEmbed,
  ]
 * *************************************************************** */

import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';

import { VideoWrapper as MapWrapper, } from '../sharedStyles/videoWrapper';
import { VideoElement as MapElement, } from '../sharedStyles/videoElement';

GoogleMap.propTypes = {
  /**
   * The type of this twitter element
   * ('static', 'search', 'directions' or 'streetView').
   */
  embedType: PropTypes.string.isRequired,
  /**
   * The address (in static map) or starting address (in direction).
   */
  content: PropTypes.string.isRequired,
  settings: PropTypes.shape({
    /**
     * The language of display.
     */
    language: PropTypes.string.isRequired,
    /**
     * Should the map be displayed as satellite map ('false' || 'true').
     */
    satellite: PropTypes.string.isRequired,
    /**
     * The finish address in Direction Map.
     */
    destination: PropTypes.string,
    /**
     * Checkpoints to include in the route (separated with '|') in Direction Map
     * (for example: 'dizengoff center | Tony Vespa').
     */
    waypoints: PropTypes.string,
    /**
     * The method of travel ('driving', 'walking', 'flying', etc) Direction Map.
     */
    mode: PropTypes.string,
    /**
     * The distance units to display ('metric' || 'imperial') Direction Map.
     */
    units: PropTypes.string,
    /**
     * The map coordinates in street view.
     */
    coordinates: PropTypes.string,
    /**
     * Camera's direction in street view.
     */
    heading: PropTypes.string,
    /**
     * Camera's vertical angle in street view (Between -90 to 90, default 0).
     */
    pitch: PropTypes.string,
    /**
     * Camera's zoom (field of view) in street view (Between 10 to 100).
     */
    fov: PropTypes.string,
  }).isRequired,
  /**
   * A function to be called when the map finishes to load.
   */
  onLoadCallback: PropTypes.func,
};

GoogleMap.defaultProps = {
  onLoadCallback: null,
};

const mapCover = () => ({
  bottom: '0',
  left: '0',
  right: '0',
  top: '0',
  cursor: 'pointer',
  position: 'absolute',
  zIndex: '1',
});

const MapCover = createComponent(mapCover, 'div', props => Object.keys(props));

function GoogleMap(props) {
  const settings = props.settings;
  const type = props.embedType;
  const address = props.content;
  const key = 'AIzaSyBIAxVLUwr1Lls-usIxb0HxCRpCXMhJtlU';

  let waypoints;
  let method;
  let units;
  let heading;
  let fov;
  let pitch;

  const zoom =
    type !== 'streetView'
      ? settings.zoom ? `&zoom=${settings.zoom}` : ''
      : '';
  const satellite =
    type !== 'streetView' && settings.satellite ? '&maptype=satellite' : '';

  const searchString =
    type === 'static'
      ? `place?q=${address}`
      : type === 'search'
        ? `search?q=${address}`
        : type === 'directions'
          ? `directions?origin=${address}&destination=${settings.destination}`
          : type === 'streetView'
            ? `streetview?location=${settings.coordinates}`
            : '';

  if (type === 'directions') {
    waypoints = settings.waypoints
      ? `&waypoints=("${settings.waypoints}")`
      : '';
    method = `&mode=${settings.mode}`;
    units = `&units=${settings.units}`;
  }
  else if (type === 'streetView') {
    heading =
      settings.heading && settings.heading !== 'none'
        ? `&heading=${settings.heading}`
        : '';
    fov = `&fov=${settings.fov}`;
    pitch = `&pitch=${settings.pitch}`;
  }

  const removeCover = e => {
    e.target.remove();
  };

  return (
    <MapWrapper>
      <MapElement
        width="600"
        height="450"
        src={`https://www.google.com/maps/embed/v1/${searchString}&key=${key}&language=${settings.language}${waypoints ||
          ''}${method || ''}${units || ''}${satellite || ''}${fov ||
          ''}${heading || ''}${zoom || ''}${pitch || ''}`}
        frameBorder="0"
        allowFullScreen=""
        onLoad={props.onLoadCallback}
      />

      <MapCover onClick={removeCover} />
    </MapWrapper>
  );
}

export default GoogleMap;
