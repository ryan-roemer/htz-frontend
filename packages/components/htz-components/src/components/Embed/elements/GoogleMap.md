<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [GoogleMap component](#googlemap-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### GoogleMap component

An example for Google map street view.

```jsx static
<GoogleMap
  embedType = 'streetView'
  settings = {{
    heading: 'none',
    coordinates: '32.05216,34.772368',
    language: 'en',
    satellite: 'false',
    pitch: '00',
    fov: '20',
  }}
  caption = 'street view'
  credit = 'GoogleMap'
/>
```
