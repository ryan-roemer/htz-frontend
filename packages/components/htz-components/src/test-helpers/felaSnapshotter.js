import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
