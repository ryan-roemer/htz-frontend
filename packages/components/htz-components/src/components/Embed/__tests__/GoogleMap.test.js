import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import GoogleMap from '../elements/GoogleMap';

it('Google-Maps static ', () => {
  const snapshot = felaSnapshotter(
    <GoogleMap
      source="shoken 21 tel aviv"
      caption="static - satellite - eng"
      credit="static - satellite - eng"
      embedType="static"
      kind="embed"
      settings={{
        language: 'en',
        satellite: 'true',
      }}
      inputTemplate="com.polobase.GoogleMapEmbed"
      contentId="7.4456"
      contentName="static - satellite - eng"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Google-Maps directions ', () => {
  const snapshot = felaSnapshotter(
    <GoogleMap
      source="shoken 21 tel aviv"
      caption="shoken 21 tel aviv ---> tony vespa habima tel aviv"
      credit="directions - noSatellite - eng"
      embedType="directions"
      kind="embed"
      settings={{
        mode: 'walking',
        destination: 'tony vespa habima tel aviv',
        language: 'en',
        satellite: 'false',
        units: 'metric',
        waypoints: 'סורה מארה',
      }}
      inputTemplate="com.polobase.GoogleMapEmbed"
      contentId="7.4458"
      contentName="directions - noSatellite - eng"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Google-Maps search ', () => {
  const snapshot = felaSnapshotter(
    <GoogleMap
      source="קופיקס בתל אביב"
      caption="קופיקס בתל אביב"
      credit="search- satellite - eng"
      embedType="search"
      kind="embed"
      settings={{
        language: 'en',
        satellite: 'true',
      }}
      inputTemplate="com.polobase.GoogleMapEmbed"
      contentId="7.4460"
      contentName="search- satellite - eng"
    />
  );
  expect(snapshot).toMatchSnapshot();
});

it('Google-Maps street view ', () => {
  const snapshot = felaSnapshotter(
    <GoogleMap
      source="streetView"
      caption="street view"
      credit="street view"
      embedType="streetView"
      kind="embed"
      settings={{
        heading: 'none',
        coordinates: '32.05216,34.772368',
        language: 'en',
        satellite: 'false',
        pitch: '00',
        fov: '20',
      }}
      inputTemplate="com.polobase.GoogleMapEmbed"
      contentId="7.4963"
      contentName="street view"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
