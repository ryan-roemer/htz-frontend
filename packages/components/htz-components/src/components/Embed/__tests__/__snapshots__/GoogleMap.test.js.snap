// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google-Maps directions  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen=""
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        height="450"
        onLoad={null}
        src="https://www.google.com/maps/embed/v1/directions?origin=shoken 21 tel aviv&destination=tony vespa habima tel aviv&key=AIzaSyBIAxVLUwr1Lls-usIxb0HxCRpCXMhJtlU&language=en&waypoints=(\\"סורה מארה\\")&mode=walking&units=metric&maptype=satellite"
        width="600"
      />
      <div
        className="htz-o htz-j htz-p htz-k htz-q htz-l htz-r"
        onClick={[Function]}
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0rem}.htz-b{padding-bottom:56.25%}.htz-c{height:0rem}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:}.htz-g{padding:0rem}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0rem}.htz-k{top:0rem}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}.htz-o{bottom:0rem}.htz-p{right:0rem}.htz-q{cursor:pointer}.htz-r{z-index:1}",
}
`;

exports[`Google-Maps search  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen=""
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        height="450"
        onLoad={null}
        src="https://www.google.com/maps/embed/v1/search?q=קופיקס בתל אביב&key=AIzaSyBIAxVLUwr1Lls-usIxb0HxCRpCXMhJtlU&language=en&maptype=satellite"
        width="600"
      />
      <div
        className="htz-o htz-j htz-p htz-k htz-q htz-l htz-r"
        onClick={[Function]}
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0rem}.htz-b{padding-bottom:56.25%}.htz-c{height:0rem}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:}.htz-g{padding:0rem}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0rem}.htz-k{top:0rem}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}.htz-o{bottom:0rem}.htz-p{right:0rem}.htz-q{cursor:pointer}.htz-r{z-index:1}",
}
`;

exports[`Google-Maps static  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen=""
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        height="450"
        onLoad={null}
        src="https://www.google.com/maps/embed/v1/place?q=shoken 21 tel aviv&key=AIzaSyBIAxVLUwr1Lls-usIxb0HxCRpCXMhJtlU&language=en&maptype=satellite"
        width="600"
      />
      <div
        className="htz-o htz-j htz-p htz-k htz-q htz-l htz-r"
        onClick={[Function]}
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0rem}.htz-b{padding-bottom:56.25%}.htz-c{height:0rem}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:}.htz-g{padding:0rem}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0rem}.htz-k{top:0rem}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}.htz-o{bottom:0rem}.htz-p{right:0rem}.htz-q{cursor:pointer}.htz-r{z-index:1}",
}
`;

exports[`Google-Maps street view  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen=""
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        height="450"
        onLoad={null}
        src="https://www.google.com/maps/embed/v1/streetview?location=32.05216,34.772368&key=AIzaSyBIAxVLUwr1Lls-usIxb0HxCRpCXMhJtlU&language=en&fov=20&pitch=00"
        width="600"
      />
      <div
        className="htz-o htz-j htz-p htz-k htz-q htz-l htz-r"
        onClick={[Function]}
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0rem}.htz-b{padding-bottom:56.25%}.htz-c{height:0rem}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:}.htz-g{padding:0rem}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0rem}.htz-k{top:0rem}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}.htz-o{bottom:0rem}.htz-p{right:0rem}.htz-q{cursor:pointer}.htz-r{z-index:1}",
}
`;
